lib/deimos/batch_consumer.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+require 'deimos/avro_data_decoder'
+require 'deimos/base_consumer'
+require 'phobos/batch_handler'
+
+module Deimos
+  # Class to consume batches of messages in a topic
+  # Note: According to the docs, instances of your handler will be created
+  # for every incoming batch of messages. This class should be lightweight.
+  class BatchConsumer < BaseConsumer
+    include Phobos::BatchHandler
+
+    # :nodoc:
+    def around_consume_batch(payloads, metadata)
+      _received_batch(payloads, metadata)
+      benchmark = Benchmark.measure do
+        _with_error_span(payloads, metadata) { yield }
+      end
+      _handle_success(benchmark.real, payloads, metadata)
+    end
+
+    # :nodoc:
+    def before_consume_batch(batch, metadata)
+      _with_error_span(batch, metadata) do
+        metadata[:keys] = batch.map do |message|
+          decode_key(message.key)
+        end
+
+        batch.map do |message|
+          self.class.decoder.decode(message.payload) if message.payload.present?
+        end
+      end
+    end
+
+    # Consume a batch of incoming messages.
+    # @param _payloads [Array<Phobos::BatchMessage>]
+    # @param _metadata [Hash]
+    def consume_batch(_payloads, _metadata)
+      raise NotImplementedError
+    end
+
+  protected
+
+    def _received_batch(payloads, metadata)
+      Deimos.config.logger.info(
+        message: 'Got Kafka batch event',
+        message_ids: _payload_identifiers(payloads, metadata),
+        metadata: metadata.except(:keys)
+      )
+      Deimos.config.logger.debug(
+        message: 'Kafka batch event payloads',
+        payloads: payloads
+      )
+      Deimos.config.metrics&.increment(
+        'handler',
+        by: metadata['batch_size'],
+        tags: %W(
+          status:batch_received
```

Comment:
Taking another look at this... this doesn't report the *number of batches* - it still reports the *number of records*. I can see value in reporting both of them. Can we increment both the standard `received` (by batch size) and `batch_received` (by 1)?